sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py

```diff
@@ -192,6 +198,7 @@ def compute(
         Y,
         k,
         metric="euclidean",
+        precomputed_matrix = None,
```

Comment:
Can you move it bellow `X` and `Y`, make their default value `None` and check in the implementation that `X == Y == None` xor `precomputed_matrix == None`?